### PR TITLE
don't build tls12-download unless it's needed

### PR DESCRIPTION
**Pull request by strega-nil**
Nobody needs to build it except us, so there's no need to.
Additionally, on mingw, tls12-download fails to build due,
so this also fixes our build on mingw.

Alternative fix to #31

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,36 +14,60 @@ include(CMakeDependentOption)
...
-file(GLOB VCPKGLIB_BASE_INCLUDES CONFIGURE_DEPENDS include/vcpkg/base/*.h include/vcpkg/base/fwd/*.h)
-
-set(VCPKG_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/src/vcpkg.cpp)
-
-file(GLOB VCPKG_TEST_SOURCES CONFIGURE_DEPENDS src/vcpkg-test/*.cpp)
-file(GLOB VCPKG_TEST_INCLUDES CONFIGURE_DEPENDS include/vcpkg-test/*.h)
-
-file(GLOB VCPKG_FUZZ_SOURCES CONFIGURE_DEPENDS src/vcpkg-fuzz/*.cpp)
+file(GLOB VCPKGLIB_SOURCES CONFIGURE_DEPENDS
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/vcpkg/*.cpp"
```
Comment (ras0219-msft) [inline CMakeLists.txt:43]: Why do these need to be qualified with `CMAKE_CURRENT_SOURCE_DIR` (is it clang-format)? If it is optional, I would prefer to omit it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -17,36 +14,60 @@ include(CMakeDependentOption)
 option(BUILD_TESTING "Option for enabling testing" ON)
 option(VCPKG_ALLOW_APPLE_CLANG "Option for allowing apple clang, even versions that we don't know will work" OFF)
 option(VCPKG_DEVELOPMENT_WARNINGS "Option for turning on all warnings" ON)
-option(VCPKG_WARNINGS_AS_ERRORS "Set warnings to be errors" ${VCPKG_DEVELOPMENT_WARNINGS})
+option(VCPKG_WARNINGS_AS_ERRORS "Set warnings to be errors" OFF)
+option(VCPKG_BUILD_TLS12_DOWNLOADER "Enable building the tls12-downloader" OFF)
 option(VCPKG_BUILD_FUZZING "Option for enabling vcpkg-fuzz support" OFF)
 option(VCPKG_EMBED_GIT_SHA "Option for to fill in the Git SHA version; off by default to avoid privacy concerns out of official builds" OFF)
 
+
```
Comment (ras0219-msft) [inline CMakeLists.txt:22]:
```suggestion
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -17,36 +14,60 @@ include(CMakeDependentOption)
...
-endif()
 if(DEFINE_DISABLE_METRICS OR VCPKG_DISABLE_METRICS)
     message(DEPRECATION "DEFINE_DISABLE_METRICS / VCPKG_DISABLE_METRICS are now handled by creating a "
         "file vcpkg.disable_metrics next to the binary.")
 endif()
 
+if(VCPKG_BUILD_TLS12_DOWNLOADER AND NOT WIN32)
+    message(FATAL_ERROR "The tls12-downloader binary is only supported on Windows.")
+endif()
+
```
Comment (ras0219-msft) [inline CMakeLists.txt:34]:
```suggestion
```
This seems unnecessarily pedantic; it's off-by-default.

**Review by ras0219-msft — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (strega-nil) [reply CMakeLists.txt:43]: I mostly wanted it to make it similar to the `set` input file commands, it's not necessary.

File: CMakeLists.txt (hunk 3 above)
Comment (strega-nil) [reply CMakeLists.txt:34]: sure, wfm.

**Pull request merged**